sos/plugins/etcd.py

```diff
@@ -19,10 +20,14 @@ class etcd(Plugin, RedHatPlugin):
     plugin_name = 'etcd'
     packages = ('etcd',)
     profiles = ('container', 'system', 'services', 'cluster')
-
-    cmd = 'etcdctl'
+    files = ('/etc/origin/node/pods/etcd.yaml',)
 
     def setup(self):
+        if path.exists('/etc/origin/node/pods/etcd.yaml'):
+            self.cmd = 'master-exec etcd etcd etcdctl'
+        else:
+            self.cmd = 'etcdctl'
```

Comment:
In theory this should just work; however, some RHEL versions use etcdv2 and others use etcdv3, and this command works using an env variable to determine which subcommand (etcdctl2 or etcdctl3) you want to use. 

I wonder if/how we could better determine what db schema etcd was using (outside of these commands) so we know what command to properly call? Without this, we might need to try calling both to see which one works?

Reply:
I don't think we need to version-control the command we use since the `etcd` package drops `/usr/bin/etcdctl` regardless of which version that's installed. The package is responsible for this, so sos shouldn't attempt to circumvent it even if it is in an attempt to ensure versioning.

Reply:
The `self.cmd` member seems to be used in just one place in exactly one method (`setup()`): let's ditch it and just use a local variable instead? (`etcd_cmd` would be a bit more readable and no great burden since we lose the `self.` prefix for e.g.).

Reply:
ack.